appleauth/key_helper.go

```diff
@@ -17,46 +17,19 @@ func fetchPrivateKey(privateKeyURL string) ([]byte, string, error) {
 		return nil, "", err
 	}
 
-	key, err := copyOrDownloadFile(fileURL)
+	// Download or load local file
+	filedownloader := filedownloader.New(http.DefaultClient)
+	fileProvider := input.NewFileProvider(filedownloader)
+	localFile, err := fileProvider.LocalPath(fileURL.String())
 	if err != nil {
 		return nil, "", err
 	}
-
-	return key, getKeyID(fileURL), nil
-}
-
-func copyOrDownloadFile(u *url.URL) ([]byte, error) {
-	// if file -> copy
-	if u.Scheme == "file" {
-		b, err := ioutil.ReadFile(u.Path)
-		if err != nil {
-			return nil, err
-		}
-
-		return b, err
-	}
-
-	// otherwise download
-	resp, err := http.Get(u.String())
+	key, err := ioutil.ReadFile(localFile)
```

Comment:
NOTE: this way we do an extra file write in case of remote key URL:
previously: download + read
now: download + write to file + read

Reply:
👍 You are right, but I would rather keep this as is and extend the input package with this functionality of providing the contents of file both for local and remote. I'll create a follow-up ticket for us.